feat(service): add healthchecks to evolution-api service

desarrollonextpro opened the pull request:
## Changes

- feat(service): add healthchecks for all evolution-api services

desarrollonextpro committed:
fix(templates): add health checks for Evolution API, Redis, and Postgres in compose config

desarrollonextpro committed:
Merge branch 'v4.x' into v4.x

ShadowArcanist commented on templates/compose/evolution-api.yaml:
```diff
@@ -11,11 +11,14 @@ services:
     image: 'evoapicloud/evolution-api:latest' # Change to specific version if needed.
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

```suggestion
    image: 'evoapicloud/evolution-api:v2.3.4'
```

ShadowArcanist commented on templates/compose/evolution-api.yaml:
```diff
@@ -158,6 +176,15 @@ services:
+
 volumes:
   evolution_instances: null
   evolution_redis: null
```
Coolify adds these automatically so no need to specify them here
```suggestion
```

ShadowArcanist commented on templates/compose/evolution-api.yaml:
```diff
@@ -11,11 +11,14 @@ services:
     image: 'evoapicloud/evolution-api:latest' # Change to specific version if needed.
     restart: always
```
Coolify adds these automatically so no need to specify them here
```suggestion
```

ShadowArcanist commented on templates/compose/evolution-api.yaml:
```diff
@@ -137,13 +140,28 @@ services:
   redis:
     image: 'redis:latest'
     command: "redis-server --port 6379 --appendonly yes\n"
     restart: always
```
Coolify adds these automatically so no need to specify them here
```suggestion
```

ShadowArcanist commented on templates/compose/evolution-api.yaml:
```diff
@@ -158,6 +176,15 @@ services:
     restart: always
```
Coolify adds these automatically so no need to specify them here
```suggestion
```

ShadowArcanist reviewed (changes requested):
The app works fine, though the Compose file needs small changes for better maintainability.

Also you have to target the PR to `next` branch instead of `v4.x`

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

desarrollonextpro committed:
Merge branch 'v4.x' into v4.x

github-actions[bot] committed:
docs: update changelog

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

github-actions[bot] committed:
docs: update changelog

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

github-actions[bot] committed:
docs: update changelog

ShadowArcanist commented:
Just a tip: please don't implement suggestions from coderabbit blindly.

On this [comment](https://github.com/coollabsio/coolify/pull/6607#discussion_r2414111339) it suggested to use `curl` for healthcheck:
<img width="961" height="751" alt="image" src="https://github.com/user-attachments/assets/72f701e1-766b-485e-a22a-2d421a54658e" />

But the container doesn't have `curl` so the healthcheck fails
<img width="1260" height="787" alt="image" src="https://github.com/user-attachments/assets/b6c455b3-73b0-43eb-8a8d-bbad7cf379d7" />

ShadowArcanist commented on templates/compose/evolution-api.yaml:
```diff
@@ -155,10 +171,15 @@ services:
+      retries: 20
+      start_period: 10s
+
   postgres_data: null
```
This line makes the compose invalid!
<img width="1505" height="499" alt="image" src="https://github.com/user-attachments/assets/25c6471a-f89c-43d4-ac02-12892f95624a" />


```suggestion
```

desarrollonextpro committed:
Update templates/compose/evolution-api.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

github-actions[bot] committed:
docs: update changelog

desarrollonextpro committed:
fix(templates): add health checks for Evolution API, Redis, and Postgres

- Use Node http healthcheck (no curl dependency)
- Pin redis:7-alpine
- Add pg_isready healthcheck
- Remove redundant restart/volumes lines
- Keep changelog scoped only to this PR

github-actions[bot] committed:
docs: update changelog

Cinzya commented on templates/compose/evolution-api.yaml:
```diff
@@ -155,10 +176,18 @@ services:
       - 'POSTGRES_PASSWORD=${SERVICE_PASSWORD_POSTGRES}'
       - 'POSTGRES_DB=${POSTGRES_DB:-postgres}'
       - 'AUTHENTICATION_API_KEY=${SERVICE_PASSWORD_AUTHENTICATIONAPIKEY}'
-    restart: always
     volumes:
       - 'postgres_data:/var/lib/postgresql/data'
+    healthcheck:
+      test:
+        - CMD-SHELL
+        - 'pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB} -h 127.0.0.1'
+      interval: 5s
+      timeout: 3s
+      retries: 20
+      start_period: 10s
+
 volumes:
-  evolution_instances: null
-  evolution_redis: null
-  postgres_data: null
+  evolution_instances:
+  evolution_redis:
+  postgres_data:
```
this section is in general not needed. Coolify adds it by itself when it parses the compose file.

peaklabs-dev committed:
Merge branch 'v4.x' into v4.x

github-actions[bot] committed:
docs: update changelog

peaklabs-dev committed:
Update CHANGELOG.md

github-actions[bot] committed:
docs: update changelog

peaklabs-dev committed:
Merge branch 'next' into v4.x

github-actions[bot] committed:
docs: update changelog

peaklabs-dev commented:
Thanks for the PR. I will update it to the latest version, format the file correctly and remove some unused stuff.
